File: Google Kick Start Problems/Perfect Subarray/PerfectSubarray.py
import math

# can fix TLE with python implementation?

T = int(input())
for i in range(1, T+1):
    l = int(input())
    arr = list(map(float, input().split()))
    if l == 1:
        num = arr[0]
        if num >= 0:
            rt = math.sqrt(num)
            if rt == int(rt):
                print(f'Case #{i}: 1')
        else:
            print(f'Case #{i}: 0')  
    else:
        total = 0
        for ptr1 in range(0, len(arr)):
            sum = arr[ptr1]
            for ptr2 in range(ptr1, len(arr)):
                if ptr2 == ptr1:
                    if sum >= 0:
                        num = math.sqrt(sum)
                    else:
                        sum = -1
                else:
                    sum += arr[ptr2]
                    if sum >= 0:
                        num = math.sqrt(sum)
                    else:
                        sum = -1
                if sum != -1 and num == int(num):
                    total += 1
        print(f'Case #{i}: {total}')

File: Google Kick Start Problems/Perfect Subarray/PerfectSubarray.java
import java.util.*;

public class PerfectSubarray {
	
	// google kick start round c, not done
	
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		int T = s.nextInt();
		for (int i = 1; i < T + 1; i++) {
			int len = s.nextInt();
			int[] arr = new int[len];
			for (int j = 0; j < len; j++) {
				arr[j] = s.nextInt();
			}
			if (len == 1) {
				double num = Math.sqrt((double) arr[0]);
				System.out.println("Case #" + i + ": " + (num == (int) num ? "1" : "0"));
			} else {
				int total = 0;
				for (int ptr1 = 0; ptr1 < len; ptr1++) {
					double sum = arr[ptr1];
					for (int ptr2 = ptr1; ptr2 < len; ptr2++) {
						double num;
						if (ptr2 == ptr1) {
							num = Math.sqrt((double) arr[ptr1]);
						} else {
							sum += arr[ptr2];
							num = Math.sqrt(sum);
						}
						if ((int) num == num) {
							total++;
						}
					}
				}
				System.out.println("Case #" + i + ": " + total);
			}
		}
	}
}
